### PR TITLE
Release 1.0.2

### PR DESCRIPTION
**Pull request by mbayerPK**
See commit messages for details

### CODE REVIEW

File: detekt.yml
Comment (PKJosh) [inline detekt.yml:1]: Let's have that match this, staying consistent with the latest conventions.

https://bitbucket.org/pkware-engineering/conventions/src/main/src/main/resources/detekt.yml

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [versions]
 junitVersion = "5.8.2"
 detektVersion = "1.20.0"
-ktlintVersion = "10.2.1"
 log4jVersion = "2.18.0"
+ktlintVersion = "11.3.2"
```
Comment (PKJosh) [inline gradle/libs.versions.toml:4]: Why not `11.5.0`?

File: detekt.yml
Comment (PKJosh) [reply detekt.yml:1]: Public repo or no, I think we'd like to enforce consistent Kotlin code across the organization, and this is still a PK repository.

For example, the fact that this commit has _removed_ trailing commas seems like a regression IMO.

Comment (mbayerPK) [reply detekt.yml:1]: that came from ktlint

Comment (PKJosh) [reply detekt.yml:1]: Yes, it did but that doesn't really change my point. Static imports, pkware-specific rules. All of it.

Comment (mbayerPK) [reply detekt.yml:1]: compare them to see

Comment (mbayerPK) [reply detekt.yml:1]: The rest of the things removed from the conventions repo detekt.yml should not ever apply to this project though

**Comment by mbayerPK** [comment]
oddly enough i updated ktlint and ran ktlintformat again and it seemd to not add the trialing commas back :(

**Comment by PKJosh** [comment]
> oddly enough i updated ktlint and ran ktlintformat again and it seemd to not add the trailing commas back :(

 Check `~/build.gradle.kts`: 
 
 `alias(libs.plugins.ktlint) apply false` ;)

**Comment by mbayerPK** [comment]
so yeah if i take that off and re-run ktlintformat it actually removes a trailing comma from settings.gradle.kts lol

**Comment by PKJosh** [comment]
> so yeah if i take that off and re-run ktlintformat it actually removes a trailing comma from settings.gradle.kts lol


Set this in `~/.editorconfig`:        `ij_kotlin_allow_trailing_comma = true`

File: .editorconfig [hunk 2]
```diff
@@ -10,6 +10,7 @@ tab_width = 4
 
 # ktlint
 ij_kotlin_imports_layout = *,java.**,javax.**,kotlin.**,^
+ij_kotlin_allow_trailing_comma = true
```
Comment (PKJosh) [inline .editorconfig:13]: Weird. Setting this for me locally caused ktlint failures when running from command line...

Comment (PKJosh) [reply .editorconfig:13]: I would have expected that you'd also see failures. And I think I see why: 

> Task :file-attribute-caching:ktlintMainSourceSetCheck FROM-CACHE

Comment (PKJosh) [reply .editorconfig:13]: For future reference, it was actually because of the need for `ij_kotlin_allow_trailing_comma_on_call_site = true`

**Review by PKJosh — APPROVED** [review]
- [ ] Squash fixups into original commit

**Review by MatthewFoster624 — APPROVED** [review]

File: file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt [hunk 3]
```diff
@@ -578,12 +574,87 @@ class AttributeCachingFileSystemTests {
...
+
+                val newAttributeView = Files.getFileAttributeView(cachingPath, AclFileAttributeView::class.java)
+                val newOwner = newAttributeView.owner
+                val newAclEntries = newAttributeView.acl
+
+                assertThat(newAttributeView).isInstanceOf(AclFileAttributeView::class.java)
+                // owner doesn't change because we didn't modify the owner in the concurrent file path
+                assertThat(originalOwner).isEqualTo(newOwner)
+                // acl entries do change because we modified the acl entries in the concurrent file path
+                assertThat(originalAclEntries).doesNotContain(newAclEntries)
```
Comment (PKJosh) [inline file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt:645]: This doesn't really seem to prove out `cached acl attributes are modified by concurrent operation using aclFileAttributeView`. This simply proves `cached paths do not cache attribute views`.

I think a better place to document that is in the project README, though the test is fine at demonstrating that.

**Review by PKJosh — APPROVED** [review]

**Pull request merged**